src/view/components/EditLoanEntry/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { LoanEntry } from '../../../store/modules/loan/types';
+import { Route } from '../../navigation/routes';
+
+import { MapStateToProps, MapDispatchToProps, connect } from 'react-redux';
+import navActions from '../../../store/modules/nav/actions';
+import { InjectedFabProps } from '../../hoc/PageComposer/FabPage/index';
+import Composer from '../../hoc/PageComposer/index';
+import { State, ThunkUpdateRow, StoreRow } from '../../../store/types';
+
+import loanThunks from '../../../store/modules/loan/thunks';
+
+import Styles from './style';
+import { getActiveFarmer } from '../../../store/modules/farmer/selectors';
+import { getActiveLoanEntry } from '../../../store/modules/loan/selectors';
+import * as moment from 'moment';
+
+
+interface OwnPropsType {
+}
+
+/** EditLoanEntry DispatchPropsType */
+interface DispatchPropsType {
+  navigate(route: Route): void;
+  goBack(): void;
+  updateLoanEntry(newEntry: ThunkUpdateRow<LoanEntry>): void;
+}
+
+/** EditLoanEntry StorePropsType */
+interface StorePropsType {
+  farmer: Farmer;
+  loanEntry: StoreRow<LoanEntry>;
+}
+
+type NestedPropsType = StorePropsType & DispatchPropsType & OwnPropsType;
+
+/** EditLoanEntry PropsType */
+type PropsType = InjectedFabProps & NestedPropsType;
+
+/** EditLoanEntry OwnStateType */
+interface OwnStateType {
+  amountOfLoan: number;
+}
+
+/**
+ * Button color
+ */
+type ButtonColor = 'PRIMARY' | 'INFO';
+
+/**
+ * Page for EditLoanEntry
+ */
+class EditLoanEntry extends React.Component<PropsType, OwnStateType> {
+
+  constructor(props: PropsType) {
+    super(props);
+    this.state = {
+      amountOfLoan: this.props.loanEntry.amount,
+    };
+  }
+
+  /** Create page buttons */
+  private renderCancelButton = () => this.renderButton('Cancel', 'INFO', this.onCancelPress);
+  private renderSaveButton = () => this.renderButton('Save', 'PRIMARY', this.onSavePress);
+
+  /** Handle pressing cancel button */
+  private onCancelPress = () => this.props.goBack();
+
+  /** Handle pressing save button */
+  private onSavePress = () => {
+    let newEntry: ThunkUpdateRow<LoanEntry> = {
+      uuid: this.props.loanEntry.uuid,
+      toPersonUuid: this.props.loanEntry.toPersonUuid,
+      fromPersonUuid: this.props.loanEntry.fromPersonUuid,
+      amount: this.state.amountOfLoan,
+      currency: 'UGX',
+    };
+    this.props.updateLoanEntry(newEntry);
+    this.props.navigate(Route.FARMER);
+  }
+
+  /**
+   * Handle entry changes, update local state
+   */
+  private onAmountChange = (newAmount: string) => this.setState(state => ({ amountOfLoan: parseFloat(newAmount) }));
+
+  /**
+   * Returns a button with text, color, and onPress callback specified
+   */
+  private renderButton(text: string, color: ButtonColor, onPress: any) {
+    const isInfo = color === 'INFO';
+    const isPrimary = color === 'PRIMARY';
+
+    return (
+      <Col style={Styles.button}>
+        <Button block info={isInfo} primary={isPrimary} onPress={onPress}>
```

Comment:
The button shouldn't be press-able unless all inputs are valid. This goes back to input validation.

Reply:
Updated. Button to add is now disabled if input is invalid